### ISSUE TITLE
Add a project-level milestones array

### ISSUE THREAD

**Issue by duncandewhurst** [open-contracting/infrastructure#253]
To allow publishers to disclose key project dates for which there isn't a specific field in OC4IDS, e.g. the operation and maintenance end date discussed in #252

**Comment by duncandewhurst** [comment]
SIF's SOURCE platform also includes the following project-level milestones which are not covered by explicit fields in OC4IDS:

* Planning start (target and actual)
* Planning end (target and actual)

ANZIP includes the following project-level milestones which are not covered by explicit fields in OC4IDS

* Construction commencement (expected and actual)

**Comment by duncandewhurst** [comment]
Example (shared in CRM-6710):

```json
{
  "projects": [
    {
      "milestones": [
        {
          "id": "1",
          "title": "Operation and Maintenance Period End Date",
          "description": "The end of the technical operation, maintenance and/or provision of services relating to the infrastructure asset",
          "dueDate": "2030-03-14T17:00:00Z",
        },
        {
        }
      ]
    }
  ]
}
```

**Comment by jpmckinney** [comment]
Closed by #426 (and again in #456)
